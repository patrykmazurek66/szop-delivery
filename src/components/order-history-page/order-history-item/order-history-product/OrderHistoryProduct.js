import "./orderhistoryproduct.css";

export default function OrderHistoryProduct({ name, qty, cost }) {
  return (
    <div className="order-history-product--container">
      <img
        className="order-history-product--img"
        src={
          "/food/thumbnails/" + name.replaceAll(" ", "-").replaceAll("'", "").toLowerCase() + ".jpg"
        }
        alt={name}
      />
      <div className="order-history-product--name-qty">
        <p className="order-history-product--name">{name}</p>
        <p className="order-history-product--quantity">Price: {(cost / qty).toFixed(2)}$</p>
        <p className="order-history-product--quantity">Quantity: {qty}</p>
      </div>
      <div className="order-history-product--cost">
        <p>{cost.toFixed(2)}$</p>
      </div>
    </div>
  );
}
