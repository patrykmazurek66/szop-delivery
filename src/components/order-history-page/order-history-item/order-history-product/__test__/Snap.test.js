import renderer from "react-test-renderer";
import OrderHistoryProduct from "../OrderHistoryProduct";

it("Product renders correctly", () => {
  const tree = renderer.create(<OrderHistoryProduct name={"Burger"} qty={1} cost={5} />).toJSON();
  expect(tree).toMatchSnapshot();
});
