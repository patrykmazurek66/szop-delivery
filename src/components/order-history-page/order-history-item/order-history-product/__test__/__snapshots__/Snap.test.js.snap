// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  className="order-history-product--container"
>
  <img
    alt="Burger"
    className="order-history-product--img"
    src="/food/thumbnails/burger.jpg"
  />
  <div
    className="order-history-product--name-qty"
  >
    <p
      className="order-history-product--name"
    >
      Burger
    </p>
    <p
      className="order-history-product--quantity"
    >
      Price: 
      5.00
      $
    </p>
    <p
      className="order-history-product--quantity"
    >
      Quantity: 
      1
    </p>
  </div>
  <div
    className="order-history-product--cost"
  >
    <p>
      5.00
      $
    </p>
  </div>
</div>
`;
